### ISSUE TITLE
Delete reference item

### ISSUE THREAD

**Issue by wedi** [hsoft/dupeguru#271]
My most desired feature: Please make it possible to delete a selection **including the reference** with a keystroke.

When checking the duplicates for correct matches it would be soo handy to be able to delete a reference picture, too.
Use case: I have a bloated (18 GB, these kinds of folders always grow exponentially 😜) picture folder called "unsorted" which contains many duplicates but also many very similar pictures from shoots (ebay items, family portraits etc.) so I need to check many, many duplicates manually. It would be great if I could directly delete the bad ones (blurred, bad contrast etc.) with a keystroke as I'm already opening every image so I don't need to touch these a second time.

I love the keyboard, so I know that iTunes binds Alt+Delete (⎇+⌫) to delete a picture from the library. Maybe you could assign an action "Delete all pictures in selection" to this key combination (with a warning that can be silenced).

Once again: thank you so much for your great and helpful toolset!

**Comment by pursual** [comment]
Yeah...or a way to just turn off the reference functionality altogether! It prevents me from marking the actual file I want to mark.

**Comment by hsoft** [comment]
@pursual to be sure: you can delete the reference file by putting another file of the group in the reference spot instead. This is done with the "Make selected reference" action.

**Comment by hsoft** [comment]
Having the possibility to delete the reference file of a duplicate set is a very dangerous thing to allow and I'm not sure yet if this feature was asked out of not knowing of the "Make selected reference" action.

Until there's more details, I'm putting this ticket on hold.

**Comment by glubsy** [comment]
I think having DupeGuru assume that reference files should only be the bigger ones is counter intuitive in some cases (including mine). I often consider the smaller file to be the reference.
I second that feature request of having the option to disable the reference system for files.

(in my case the "make selected reference doesn't work" by the way, opening a ticket for that)

**Comment by hsoft** [comment]
@glubsy Did you know that you can use the Re-prioritize dialog to do exactly that? Select the "Size" criteria kind and choose "Smallest".

**Comment by wedi** [comment]
@hsoft I do know the "Make selected reference“ action. As I explained in my initial post, I want to be able to delete bad pictures completely when checking for duplicates so I don’t have to touch them a second time. Maybe this action could require additional activation to make sure no one deletes something by accident.

**Comment by hsoft** [comment]
@wedi yes, I understand that you understand :) My question was directed more towards @glubsy for whom the "Make selected reference“ feature seemed like it would do the trick.

I don't know why I had applied the Hold label on the ticket though. removing.

As for me, I'm not fundamentally against something of the sort. If someone comes with a PR that a sensible implementation of the idea that doesn't jeopardize security (as in accidentally deleting stuff), I'll merge it.

**Comment by TerraTech** [comment]
(I'm using the Windows 7 version)

I too would like a (I Know What I'm Doing) 'preference' option that allows for selecting the reference file...  My usage case is with Eudora mail client (old but is rock solid with 500K+ msgs) with its attachment and embedded folders which contains a mountain of images due to spam...  Using dupeGuru to find similar images is a fast way to clean up these folders by deleting 'all' similar images...

A nice side-effect feature, or sub-preference, would be that when either:
a) you click on the reference item
b) you check its box

that it will automatically select all similar images in the group as I have already vetted the images and the whole group is safe to delete or move them elsewhere...

Preferences:
==: Allow selecting reference files (DANGEROUS)
====: Auto mark similar files as a group when reference file is selected (or clicked)
====: Allow right click operations on reference file  (**this would honor the 'Auto mark' option to allow for group actions)

As an aside, 'moving them elsewhere', I can then use a program I wrote that manipulates Eudora's MBX and TOC files directly (offline) that will use the spam images I found and look for the message itself and delete it...  This is what I've been doing with 'XnView MP' and its duplicate photo finder, but it is cumbersome to use and having the ability to use dupeGuru would remove that barrier as its similarity heuristics is pretty darn good with no false positives yet...  :)
